src/common/util/util.ts

```diff
@@ -164,16 +165,19 @@ export function* iterRange<T>(n: number, fn: (i: number) => T): Iterable<T> {
   }
 }
 
-export type TypedArrayBufferView =
-  | Uint8Array
-  | Uint8ClampedArray
-  | Uint16Array
-  | Uint32Array
-  | Int8Array
-  | Int16Array
-  | Int32Array
-  | Float32Array
-  | Float64Array;
+const TypedArrayBufferViewTypes = [
+  new Uint8Array(),
+  new Uint8ClampedArray(),
+  new Uint16Array(),
+  new Uint32Array(),
+  new Int8Array(),
+  new Int16Array(),
+  new Int32Array(),
+  new Float32Array(),
+  new Float64Array(),
+] as const;
```

Comment:
do we need to actually construct them? I think you can juts list them without new and (), and below, say result[v.name] = v;

Reply:
So I originally had that, but if I don't construct them, then

`export type TypedArrayBufferView = typeof TypedArrayBufferViewTypes[number];`

is a union of constructor function types instead of just the types. Since generating the table was easier to manipulate than modifying the `typeof`, I just changed the array to be instances so that the inferred types were correct. Not sure if there's a better way though.

Reply:
ok, idk a better way so I guess it's fine.

let's call it kTypedArrayBufferViewInstances since it's a value not a type